### ISSUE TITLE
Dynamic dispatch

### ISSUE THREAD

**Issue by lerno** [c3lang/c3c#761]
Allow dynamic dispatch on specially marked calls.

For example we may get:

```
any x = ...
x->foo(123);
```
Where `foo(123)` is a runtime lookup.

**Comment by lerno** [comment]
Possible ways to declare the function:
```
fn void any.foo(int x);
extern fn void any.foo(int x);
dynamic fn void any.foo(int x);
fn void any.foo(int x) @dynamic;
extern void any.foo(int x) @dynamic;
```

**Comment by lerno** [comment]
Implemented in #767
